go-controller/pkg/ovn/pods.go

```diff
@@ -115,7 +115,13 @@ func (oc *Controller) deleteLogicalPort(pod *kapi.Pod, portInfo *lpInfo) (err er
 		// is not re-added into the cache. Delete logical switch port anyway.
 		annotation, err := util.UnmarshalPodAnnotation(pod.Annotations)
 		if err != nil {
-			return fmt.Errorf("unable to unmarshal pod annocations for pod %s/%s: %w", pod.Namespace, pod.Name, err)
+			if util.IsAnnotationNotSetError(err) {
+				// if the annotation doesn’t exist, that’s not an error. It means logical port does not need to be deleted.
+				klog.V(5).Infof("Unable to unmarshal pod annotations for pod %s/%s: %w", pod.Namespace, pod.Name, err)
```

Comment:
Flavio, should we log here something more along the lines of `"No annotations on pod %s/%s, no need to delete its logical port: %s"`?

Reply:
sure thing, @ricky-rav . I will re-phrase it as you mentioned.